src/external/wannierio.jl

```diff
@@ -0,0 +1,693 @@
+kpoint.
+Since the `Kpoint.spin` already contains the spin information, no need
+to specify `spin`, which is needed in `compute_mmn`, `compute_eig`, etc.
+
+For the explanation of the `ψ`, see [`compute_mmn`](@ref).
+"""
+@timing function compute_unk(
+    basis::PlaneWaveBasis, ψ::AbstractVector{<:AbstractMatrix{<:Complex}}, kpt::Kpoint
+)
+    fft_size = basis.fft_size
+    ik = findfirst(Ref(kpt) .== basis.kpoints)
+    n_bands = size(ψ[ik], 2)
+    unk = zeros(eltype(ψ[ik]), fft_size..., n_bands)
+
+    for n in 1:n_bands
+        unk[:, :, :, n] = ifft(basis, kpt, @view ψ[ik][:, n])
+    end
+    unk
+end
+
+"""
+Generate a `NamedTuple` of input parameters for a Wannier90 `win` file.
+Parameters to Wannier90 can be added as kwargs: e.g. `num_iter=500`.
+"""
+function get_wannier90_win(basis::PlaneWaveBasis; kwargs...)
+    isnothing(basis.kgrid) && error("The basis must be constructed from a MP grid.")
+    # Required parameter
+    @assert :num_wann in keys(kwargs)
+    @assert :num_bands in keys(kwargs)
+
+    win = Dict{Symbol, Any}(kwargs)
+
+    # columnwise vectors, and to angstrom
+    to_ang(v) = auconvert.(Unitful.angstrom, v).val
+    unit_cell_cart = to_ang.(basis.model.lattice)
+    push!(win, :unit_cell_cart => unit_cell_cart)
+
+    atom_labels = [string(a.symbol) for a in basis.model.atoms]
+    # to columnwise vectors, and fractional coordinates
+    atoms_frac = hcat(basis.model.positions...)
+    push!(win, :atoms_frac => atoms_frac, :atom_labels => atom_labels)
+
+    # deepcopy for safety
+    mp_grid = deepcopy(basis.kgrid)
+    kpoints = filter(k -> k.spin == 1, basis.kpoints)
+    # columnwise vectors, fractional coordinates
+    kpoints = hcat([k.coordinate for k in kpoints]...)
+    @assert size(kpoints) == (3, prod(mp_grid))
+    push!(win, :mp_grid => mp_grid, :kpoints => kpoints)
+
+    if get(kwargs, :wannier_plot, false)
+        push!(win, :wvfn_formatted => true)
+    end
+
+    if get(kwargs, :bands_plot, false)
+        kpath  = irrfbz_path(basis.model)
+        kpoint_path = []
+        for path in kpath.paths
+            for i in 1:length(path)-1
+                A, B = path[i:i+1]  # write segment A -> B
+                A_k = A => kpath.points[A]
+                B_k = B => kpath.points[B]
+                push!(kpoint_path, [A_k, B_k])
+            end
+        end
+        push!(win, :kpoint_path => kpoint_path)
+    end
+
+    # Return a NamedTuple so that the user can access the parameters using
+    # win.num_wann, win.num_bands, etc. (with Dict it is win[:num_wann])
+    NamedTuple(win)
+end
+
+"""
+Preprocess the scf results for Wannierization.
+
+Unfold symmetries, exclude bands.
+
+# Arguments
+- `exclude_bands`: the band indices to be excluded from the Wannierization,
+    usually the semicore states.
+"""
+function unfold_scfres_wannier(
+    scfres::NamedTuple,
+    exclude_bands::Union{AbstractArray{<:Integer},Nothing}=nothing,
+)
```

Comment:
@LaurentVidal95 Didn't you write a function at some point that does more or less this?